packages/webapp/src/containers/CropCatalogue/useStringFilteredCrops.ts

```diff
@@ -49,8 +49,17 @@ export default function useStringFilteredCrops(
         'crop_cultivar', // crop_variety.crop_cultivar
       ];
       for (let i = 0; i < keys.length; i++) {
-        if (keys[i] in crop && check(crop[keys[i] as keyof typeof crop])) {
-          return true;
+        if (keys[i] in crop) {
+          if (!i) {
```

Comment:
@SayakaOno This works but I am not sure I fully understand.

Is this when i=1 or when keys[i] == 'crop_specie'?

Reply:
I prefer the use of the explicit if (keys[i] === 'crop_genus') since there is no way to misunderstand that. Right now there is a chance some developer will come along and want to search by "crop_notes" or something and then add that in as the first element in the keys array.

Reply:
Thanks to your input, it should be more readable now. I appreciate it!!

Reply:
ooh thats a pretty solution!